File: dotnet/Lexy/Lexy.Compiler/Language/ColumnHeader.cs
using System.Collections.Generic;
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language
{
    public class ColumnHeader : Node
    {
        public string Name { get; }
        public VariableDeclarationType Type { get; }

        public ColumnHeader(string name, VariableDeclarationType type, SourceReference reference) : base(reference)
        {
            Name = name;
            Type = type;
        }

        public static ColumnHeader Parse(string name, string typeName, SourceReference reference)
        {
            var type = VariableDeclarationType.Parse(typeName);
            return new ColumnHeader(name, type, reference);
        }

        public override IEnumerable<INode> GetChildren()
        {
            yield break;
        }

        protected override void Validate(IValidationContext context)
        {
        }
    }
}

File: typescipt/src/language/tables/columnHeader.ts


export class ColumnHeader extends Node {
   public string Name
   public VariableDeclarationType Type

   public ColumnHeader(string name, VariableDeclarationType type, SourceReference reference) {
     super(reference);
     Name = name;
     Type = type;
   }

   public static parse(name: string, typeName: string, reference: SourceReference): ColumnHeader {
     let type = VariableDeclarationType.parse(typeName, reference);
     return new ColumnHeader(name, type, reference);
   }

   public override getChildren(): Array<INode> {
     yield return Type;
   }

   protected override validate(context: IValidationContext): void {
   }
}
